core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +308,26 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput') {
+        var numChoices = $scope.getAnswerChoices().length;
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
```

Comment:
I think what Sean means is, you need to start off with an array of length equal to the number of choices in the item selection interaction, with values all false. Then, if you encounter Equals or ContainsAtLeastOneOf the item at position k, you should set the value at position k of the array to true. If you encounter DoesNotContainAtLeastOneOf, you should set all values except the one at position k to true.

Does that make sense?